tests/run/i1169b.scala

```diff
@@ -0,0 +1,18 @@
+
+object Test {
+  def main(args: Array[String]): Unit = {
+    val xProducer = new XProducer
+    println(xProducer.produce.underlying)
```

Comment:
Neither of this tests is compete.

Those tests should additionally check that `Producer[T]` behaves correctly in a generic position. E.g. try defining a `toString` in X and calling it in a generic context of `Producer[T]`, where T isn't known to be `X`.
I would be suprised if it does. 

I had a look on generated bytecode. The bug is NOT fixed and the bytecode is wrong.